lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -58,74 +61,85 @@ defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
 
   @tag [:io]
   @impl Http
-  def request_many(uris) do
-    if check_too_many_calls() do
-      raise Library.Error, message: "Http module got called more than once"
-    end
+  def request_many(requests) do
+    check_too_many_calls()
 
-    uris_count = length(uris)
+    requests_count = length(requests)
 
-    if uris_count > 5 do
+    if requests_count > 5 do
       raise Library.Error, message: "Http.request_many/1 was called with too many urls"
-    else
-      Task.Supervisor.async_stream_nolink(
-        TaskSupervisor,
-        uris,
-        &do_request/1,
-        ordered: true,
-        max_concurrency: 5,
-        timeout: @timeout,
-        on_timeout: :kill_task
-      )
-      |> Enum.zip(uris)
-      # count the number of bytes to be able to send a error too large
-      # this is sub optimal because miners might still download threshold N times before returning the error
-      # TODO: improve this
-      |> Enum.reduce({0, []}, fn
-        {{:exit, :timeout}, uri}, _ ->
-          raise Library.Error,
-            message: "Http.request_many/1 timed out for url: #{uri}"
-
-        {{:ok, {:error, :threshold_reached}}, uri}, _ ->
-          raise Library.Error,
-            message: "Http.request_many/1 response is bigger than threshold for url: #{uri}"
-
-        {{:ok, {:error, :not_https}}, uri}, _ ->
-          raise Library.Error,
-            message: "Http.request_many/1 was called with a non https url: #{uri}"
-
-        {{:ok, {:error, _}}, uri}, _ ->
-          # Mint.HTTP.connect error
-          # Mint.HTTP.stream error
-          raise Library.Error,
-            message: "Http.request_many/1 failed for url: #{uri}"
-
-        {{:ok, {:error, _, _}}, uri}, _ ->
-          # Mint.HTTP.request error
-          raise Library.Error,
-            message: "Http.request_many/1 failed for url: #{uri}"
-
-        {{:ok, {:ok, map}}, _uri}, {bytes_acc, result_acc} ->
-          bytes =
-            case map["body"] do
-              nil -> 0
-              body -> byte_size(body)
-            end
+    end
 
-          {bytes_acc + bytes, result_acc ++ [map]}
-      end)
-      |> then(fn {bytes_total, results} ->
-        if bytes_total > @threshold do
-          raise Library.Error,
-            message: "Http.request_many/1 sum of responses is bigger than threshold"
-        else
-          results
-        end
+    requests =
+      Enum.map(requests, fn request = %{"url" => url, "method" => method} ->
```

Comment:
Suggestion: you could reduce iteration by using this logic within the async_stream_nolink, like:
```elixir
Task.Supervisor.async_stream_nolink( TaskSupervisor, requests, fn request ->
   # Handle assertion and formatting
   %{ "url" => url, "method" => method, "headers" => headers, "body" => body } = prepare_request(request)
   do_request(url, method, headers, body)
end)
...
```